app/Http/Controllers/Api/AssetsController.php

```diff
@@ -624,81 +625,70 @@ public function store(StoreAssetRequest $request): JsonResponse
      * Accepts a POST request to update an asset
      *
      * @author [A. Gianotto] [<snipe@snipe.net>]
-     * @param \App\Http\Requests\ImageUploadRequest $request
      * @since [v4.0]
-     * @return \Illuminate\Http\JsonResponse
      */
-    public function update(ImageUploadRequest $request, $id)
+    public function update(UpdateAssetRequest $request, Asset $asset): JsonResponse
```

Comment:
It looks like `$id` was actually used on, what is now, line 676. Changing it from `$id` to `$asset->id` should work.

Reply:
🤦‍♂️ thank you.